### PR TITLE
Test function numeric

### PR DESCRIPTION
**Pull request by MhmdJamaal**
- Add numeric function.
- Add testing to edge cases for the function.
- Add dependencies to pyproject.toml file.

### CODE REVIEW

File: src/group_32/optimize_numeric.py [hunk 1]
```diff
@@ -60,10 +63,60 @@ def optimize_numeric(df: pd.DataFrame) -> pd.DataFrame:
...
+
+
+
+
+
+
+
+
+
+    # for col in df.select_dtypes(include=['integer']).columns:
```
Comment (Vlee98p) [inline src/group_32/optimize_numeric.py]: Please remove commented codes at the end
